common/nexus/failure.go

```diff
@@ -338,6 +350,56 @@ func ConvertGRPCError(err error, exposeDetails bool) error {
 	return err
 }
 
+// ConvertHandlerError converts a nexus.HandlerError returned by a Nexus client into a gRPC serviceerror.
+func ConvertHandlerError(err *nexus.HandlerError, exposeDetails bool) error {
```

Comment:
Do we need `exposeDetails` here? I think we already control what to expose in the HTTP handler. Let's keep the control there.